lite/examples/sound_classification/android/app/src/main/java/org/tensorflow/lite/examples/soundclassifier/MainActivity.kt

```diff
@@ -18,84 +18,151 @@ package org.tensorflow.lite.examples.soundclassifier
+  private lateinit var handler: Handler // background thread handler to run classification
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
 
     val binding = ActivityMainBinding.inflate(layoutInflater)
     setContentView(binding.root)
 
-    soundClassifier = SoundClassifier(this, SoundClassifier.Options()).also {
-      it.lifecycleOwner = this
-    }
-
     with(binding) {
       recyclerView.apply {
-        setHasFixedSize(true)
-        adapter = probabilitiesAdapter.apply {
-          labelList = soundClassifier.labelList
-        }
+        setHasFixedSize(false)
+        adapter = probabilitiesAdapter
       }
 
+      // Input switch to turn on/off classification
       keepScreenOn(inputSwitch.isChecked)
       inputSwitch.setOnCheckedChangeListener { _, isChecked ->
-        soundClassifier.isPaused = !isChecked
+        if (isChecked) startAudioClassification() else stopAudioClassification()
         keepScreenOn(isChecked)
       }
 
-      overlapFactorSlider.value = soundClassifier.overlapFactor
-      overlapFactorSlider.addOnChangeListener { _, value, _ ->
-        soundClassifier.overlapFactor = value
+      // Slider which control how often the classification task should run
+      classificationIntervalSlider.value = classificationInterval.toFloat()
+      classificationIntervalSlider.setLabelFormatter { value: Float ->
+        "${value.toInt()} ms"
       }
-    }
-
-    soundClassifier.probabilities.observe(this) { resultMap ->
-      if (resultMap.isEmpty() || resultMap.size > soundClassifier.labelList.size) {
-        Log.w(TAG, "Invalid size of probability output! (size: ${resultMap.size})")
-        return@observe
+      classificationIntervalSlider.addOnChangeListener { _, value, _ ->
+        classificationInterval = value.toLong()
+        stopAudioClassification()
+        startAudioClassification()
       }
-      probabilitiesAdapter.probabilityMap = resultMap
-      probabilitiesAdapter.notifyDataSetChanged()
     }
 
+    // Create a handler to run classification in a background thread
+    val handlerThread = HandlerThread("backgroundThread")
+    handlerThread.start()
+    handler = HandlerCompat.createAsync(handlerThread.looper)
+
+    // Request microphone permission and start running classification
     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
       requestMicrophonePermission()
     } else {
-      soundClassifier.start()
+      startAudioClassification()
     }
+
+  }
+
+  private fun startAudioClassification() {
+    // If the audio classifier is initialized and running, do nothing.
+    if (audioClassifier != null) return;
+
+    // Initialize the audio classifier
+    val classifier = AudioClassifier.createFromFile(this, MODEL_FILE)
+    val audioTensor = classifier.createInputTensorAudio()
+
+    // Initialize the audio recorder
+    val recorder = classifier.createAudioRecord()
```

Comment:
Nit: if we're calling the member variable audioRecord, should we call this just `record` for consistency (also with the type name)?